roles/haproxy/defaults/main.yml

```diff
@@ -1,4 +1,4 @@
 tls_backend_ca: /etc/pki/haproxy/backend.{{ base_domain }}.pem
-haproxy_removal_applications: []
 haproxy_cookie_max_idle: 3600
 haproxy_metricbeat: False
+haproxy_supported_http_methods: "GET HEAD OPTIONS POST PUT DELETE"
```

Comment:
Should we somehow document how we arrived at this list? Who uses what? How do we know that perhaps we can remove one method at some point?

Reply:
I'm not very sure where we should document this, and how that list should be kept up to date. How would you propose that?
Pragmatically speaking, I'd say that by whitelisting these most common methods we at least keep out any other methods like TRACE. If needed, apps can themselves keep a more restricted list. It doesn't seem very realistic that any of these methods would disapear in the near future anyway.

Reply:
Someone in the (near) future will come and ask "why these specific methods" or "why is `OPTIONS` in there". Maybe it is already sufficient to just add a comment that says that this is the full list minus things we know we never need.

Reply:
I've added a comment in the defaults.yml.